packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -109,9 +174,67 @@ const ShippingMultiAddress = ({
             }
         }
     )
-    const {data: customer} = useCurrentCustomer()
+    const {data: customer, refetch: refetchCustomer} = useCurrentCustomer()
     const addresses = customer?.addresses || []
     const [selectedAddresses, setSelectedAddresses] = useState({})
+    const [openDropdown, setOpenDropdown] = useState(null)
+    const dropdownRefs = useRef({})
+    const [focusedIndex, setFocusedIndex] = useState(-1)
+
+    // Add address form state - track which product card is showing the add address form
+    const [showAddAddressForm, setShowAddAddressForm] = useState({}) // productId-index -> boolean
+    const [isSubmitting, setIsSubmitting] = useState(false)
+    const [formErrors, setFormErrors] = useState({}) // Track form-specific errors
```

Comment:
We use react-hook-form to manage form state and error. Do we need this state? from what I noticed, you are using as an error from API call. If you move `useForm` to this component, you will have more flexibility to control the form submission state, which you can set error accordingly without using react state here